### DISCUSSION TITLE
Documentation for binding x-ref (which is not allowed)

### DISCUSSION THREAD

**Discussion by kslstn** [alpinejs/alpine#3343] [4. General]
I spent a lot of time debugging a script only to discover, via in [this comment](https://github.com/alpinejs/alpine/discussions/1527#discussioncomment-879974), that we can't bind `x-ref` in Alpine 3. 

I'd like to do a PR for the docs, but before I do, I'd like to better understand the situation. After all, this could be a comment on the pages for `x-ref`, `bind` and even `x-for`.

1. The docs [mention](https://alpinejs.dev/directives/bind#bind-directives) at least one way to set x-ref using `bind`, albeit a static way. Are there other ways?
2. Is `x-ref` the only property that can't be bound?
3. Is there a recommended, Alpine-specific way to select dynamically created elements, like in an `x-for` loop? Or is it just plain old `document.getElementById` type of stuff there?

**Comment by ekwoka** [comment]
1. No.
2. Yes (well, maybe also binding any directive like `x-bind:x-text`?)
3. You can key the elements in an object yourself `x-init="myelements[mydynamickey] = $el"` That's not unlike what x-ref does.

But in all my experience using Alpine, I'd put x-ref in there with x-html: You basically never need this for any reason other than interfacing with some other library.

If you're doing pure alpine (or at least not passing the ref into another library) then there is a much better way to do whatever you're trying to do.

(This is my educated opinion working professionally in Alpine since v2, but I am not a maintainer of alpine nor do I chat with the maintainers about the philosophy of the product)

**Comment by kslstn** [comment]
Ooh nice: `x-init="doSomethingWith($el)"` opens up many things I wanted to do, like sending that element's position to the parent! I will reconsider my ideas for the docs because of what you wrote (maybe that actually could be clearer—my experience with Alpine so far is that anything you can do declaratively is better as it reduces timing issues).

Also I should have read the docs better, because I never realized how useful `$el` could be.

**Reply by ekwoka** [reply]
It's definitely best to keep it as declarative as possible. I just don't find `x-ref` to be really more declarative than the x-init, tbh. It's just having the function abstracted or not. Basically the same thing at the end of the day.
